plugin-server/src/worker/ingestion/team-manager.ts

```diff
@@ -89,6 +89,23 @@ export class TeamManager {
         }
     }
 
+    public isAboveUsageLimit(team: Team, key: string): boolean {
+        const { usage } = team
+        if (!usage) {
+            return false
+        }
+        const usageRecord = usage[key]
+        if (!usageRecord) {
+            return false
+        }
+        const limit = usageRecord.limit
+        if (!limit) {
+            return false
+        }
+        const current = usageRecord.usage
+        return current > limit
+    }
+
```

Comment:
I'm actually wondering if we want to move this comparison logic into the actual usage record. So we save

```
recordings: {
  usage: 120000,
  limit: 100000,
  rate_limited_until: Date
}
```

This would make the logic here more straight forward and defer the decision to wherever we report the usage, rather than the plugin-server. Not 100% sure if this is the right approach but something like it...